src/app/app.component.ts

```diff
@@ -293,5 +306,16 @@ export class AppComponent {
         // this.backgroundOptionSelected.emit(selectedBackground);
 
         this.selectedBackground = selectedBackground;
+        this.selectedPrincipleName = '';
+        this.selectedPrinciple = null;
+    }
+
+    onPrincipleOptionSelect(selectedPrincipleName: string) {
+        this.selectedPrincipleName = selectedPrincipleName;
+        const principleCategoryData = principles[this.selectedBackground.principleCategory];
+        this.selectedPrinciple = new Principle(principleCategoryData[selectedPrincipleName]);
```

Comment:
I'm confused about why we need to do it this way, but we can talk about it when we meet if you like.